service/history/workflow/update/update.go

```diff
@@ -303,15 +305,17 @@ func (u *Update) OnMessage(
 	}
 }
 
-// ReadOutgoingMessages loads any outbound messages from this Update state
-// machine into the output slice provided.
-func (u *Update) ReadOutgoingMessages(out *[]*protocolpb.Message, sequencingID *protocolpb.Message_EventId) {
+// AppendOutgoingMessages loads any outbound messages from this Update state
+// machine into the output slice provided. It also links the Update with workflow task,
+// for which it is sending messages, by setting provided workflowTaskScheduledEventID.
+func (u *Update) AppendOutgoingMessages(out *[]*protocolpb.Message, workflowTaskScheduledEventID int64, sequencingID *protocolpb.Message_EventId) {
```

Comment:
I would prefer to return the result of append as a return value from the function instead of modifying the original slice. From performance point of view, it will be the same, but `a = f(a)` screams that a will be modified while in `f(&a)` one may miss &. Even worse, what if instead of `a` I have a pointer to `a`. Then the call will be `f(p)`.

Reply:
I also don't like it but tried not to touch unrelated part. End up changing this too.